features/admin.tenants.v1/components/dropdown/tenant-dropdown.tsx

```diff
@@ -230,7 +247,29 @@ const TenantDropdown: FunctionComponent<TenantDropdownInterface> = (props: Tenan
             setTempTenantAssociationsList(association.associatedTenants);
         }
         setTenantAssociations(association);
-    }, [ associatedTenants, defaultTenant, tenantDomain ]);
+    }, [ associatedTenants, defaultTenant, currentTenant ]);
+
+    useEffect(() => {
+        if (isCentralDeploymentEnabled) {
+            getDeploymentUnits()
+                .then((response: DeploymentUnitResponse) => {
+                    setDeploymentUnits(response.deploymentUnits);
+                })
+                .catch((error: any) => {
+                    dispatch(
+                        addAlert({
+                            description:
+                                error?.description &&
+                                t("tenants:listDeploymentUnits.description"),
+                            level: AlertLevels.ERROR,
+                            message:
+                                error?.description &&
```

Comment:
Using error?.description for both message and description will not look good on UI.

Reply:
This is how we implemented error cases in all places on this class. We check if the error.description value is available and print the i18 string for description and message

Reply:
Ack